Feature/awf/scenario toggle#174

CloudNiner opened the pull request:
## Overview

Replaces the scenario dropdown in the top bar above the lab chart with a toggle, which only displays RCP45 and RCP85. Historical was removed with the intention to display it as part of the chart in #173.


### Demo

![screen shot 2017-08-01 at 09 47 01](https://user-images.githubusercontent.com/1818302/28828462-d9ad48dc-769e-11e7-9462-70700aaf217c.png)
![screen shot 2017-08-01 at 09 46 55](https://user-images.githubusercontent.com/1818302/28828461-d9acfca6-769e-11e7-8212-99f30160b25a.png)


### Notes

Styling is still iffy, will need a quick pass by a designer, along with quite a few of these other issues at some point. I will open a separate issue to track follow-up designer tasks.

I began a refactor where in parent component/modules we only import from the `index.ts` file, rather than by the explicit file path to a component. This allows us to add/remove/update/move components and filenames without having to go through the tedious process of updating file paths all over the codebase. In addition, it makes it easy to see at a glance what the "public" interface of a module is by looking at that module's `index.ts` file. We should continue this practice in as many places as possible as we refactor the UI for this sprint.

## Testing Instructions

 * View a lab chart and the project add/edit page. You should see the dropdown replaced with a toggle in both cases. When the toggle is changed, the chart/project should update and request the other scenario.

Closes #174

flibbertigibbet commented:
Looks like the karma imports need updating to fix tests.

fungjj92 commented on src/app/lab/components/scenario-toggle.component.ts:
```diff
@@ -49,7 +44,7 @@ export class ScenarioDropdownComponent implements OnInit {
     private getScenarios() {
         this.scenarioService.list().subscribe(data => {
-            this.scenarios = data;
+            this.scenarios = data.filter(s => s.name === 'RCP85' || s.name === 'RCP45')
```
In the interest of constants book-keeping, consider saving constants 'RCP85' and 'RCP45' to a list and doing a `.includes(s.name)` or other check

CloudNiner commented:
Updated. I fixed up the following:
- The styling on the add/edit page
- Using includes and a declarative array rather than in place ORs for choosing the scenarios to include in the toggle
- A bug where clicking one of the toggle buttons on the add/edit page would submit the parent form

fungjj92 commented:
** When the build failure is figured out
